### PR TITLE
MAUI 395 port editor Autosize to handler

### LINKED ISSUE #3425: Editor inside VerticalStackLayout automatically grows vertically on Android/WinUI but not on iOS
### Description

Given the following XAML

```XAML
        <VerticalStackLayout Margin="12">
            <Editor AutoSize="TextChanges"  />
        </VerticalStackLayout>
```

If you enter a new line Android/WinUI will grow vertically but iOS does not

Once this is fixed uncomment the following test
https://github.com/AndrewLang/maui/blob/4901cd7e50ccef841f148f78334004001591f72a/src/Controls/tests/DeviceTests/Elements/EditorTests.cs#L14

### Steps to Reproduce

[aeditortest.zip](https://github.com/dotnet/maui/files/7548759/aeditortest.zip)

1) run app
2) type text
3) hit the "Enter" key 

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

main

### Affected platforms

iOS

### Affected platform versions

iOS 15

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by AndrewLang**
### Description of Change ###

Implements #395 Port Editor.AutoSize to handler

- This currently doesn't work on iOS until this is resolved https://github.com/dotnet/maui/issues/3425
- In .NET MAUI the nature of layouts make this property a bit awkward. By default the editor in .NET MAUI will always remeasure itself whenever the text changes and will naturally grow if it can. In XF everything only measured itself once, so once an Editor was measured it would just remain that size. This made the property a bit more relevant with XF. Whereas in .NET MAUI you'd just be inclined to set a specific height if you didn't want it to grow
- https://github.com/dotnet/maui/issues/3429

### Additions made ###

- Adds DeviceTests for initial Aut0Size values on WinUI and Android

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [x] Targets a single property for a single control (or intertwined few properties)
- [x] Adds the property to the appropriate interface
- [x] Avoids any changes not essential to the handler property
- [x] Adds the mapping to the PropertyMapper in the handler
- [x] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [x] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [x] Adds the property to the stub class
- [x] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [x] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=1202) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=1202) before we can accept your contribution.<br/><br/>:x: AndrewLang [sign now](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=1202)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/maui?pullRequest=1202) it.</sub>

**Comment by AmrAlSayed0** [comment]
The thing about this property is that it is not a inherit property of the editor but it is a property of the _behavior_ of the editor. This changes the font size of the editor, should FontSize on the VirtualView be changed to be synced with the actual editor's FontSize? I mean this is a rare case but I would imagine a scenario where someone auto sizes an editor and then reads the new size to use it to size some other editor's or label's font.

File: src/Core/tests/DeviceTests.Android/Core.DeviceTests.Android.csproj [hunk 1]
```diff
@@ -32,6 +32,9 @@
     <BundleAssemblies>false</BundleAssemblies>
     <AndroidUseSharedRuntime>false</AndroidUseSharedRuntime>
     <EmbedAssembliesIntoApk>true</EmbedAssembliesIntoApk>
+    <AotAssemblies>false</AotAssemblies>
```
Comment (jsuarezruiz) [inline src/Core/tests/DeviceTests.Android/Core.DeviceTests.Android.csproj:35]: These changes seem unrelated, could you undo them?

**Review by jsuarezruiz — COMMENTED** [review]
Could you also add the [PortHandler] attributes to the Renderers?. Thanks!

**Comment by AndrewLang** [comment]
> Could you also add the [PortHandler] attributes to the Renderers?. Thanks!

Yes, I have added attribute [PortHandler] to EditorRenderer. Anything els.

File: src/Core/src/Platform/Android/EditTextExtensions.cs [hunk 2]
```diff
@@ -217,14 +218,24 @@ public static void UpdateReturnType(this AppCompatEditText editText, IEntry entr
...
-		[PortHandler]
+		public static void UpdateAutoSize(this AppCompatEditText editText, IEditor editor)
+		{
+			if (editor.AutoSize == EditorAutoSizeOption.Disabled)
+			{
+				editText.SetAutoSizeTextTypeWithDefaults(AutoSizeTextType.None);
+			}
+			else
+			{
+				editText.SetAutoSizeTextTypeUniformWithConfiguration(editText.AutoSizeMinTextSize, editText.AutoSizeMaxTextSize, editText.AutoSizeStepGranularity, (int)ComplexUnitType.Sp);
```
Comment (PureWeen) [inline src/Core/src/Platform/Android/EditTextExtensions.cs:229]: I don't think this is the correct API to use for this one.  Doesn't this API cause the font to autosize? We don't want the font to change size based on the size of the editor.  In the original implementation Android was able to just update its size by re-measuring so we might need to change the invalidation code.

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Comment by PureWeen** [comment]
@AndrewLang I rebased this PR as well so hopefully that doesn't mess you up too much locally

File: src/Core/src/Platform/Windows/TextBoxExtensions.cs [hunk 3]
```diff
@@ -119,5 +119,17 @@ public static void UpdateMaxLength(this MauiTextBox textBox, IEntry entry)
 			if (currentControlText.Length > maxLength)
 				textBox.Text = currentControlText.Substring(0, maxLength);
 		}
+
+		public static void UpdateAutoSize(this MauiTextBox textBox, IEditor editor)
+		{
+			if (editor.AutoSize == EditorAutoSizeOption.Disabled)
+			{
+				textBox.AcceptsReturn = false;
```
Comment (PureWeen) [inline src/Core/src/Platform/Windows/TextBoxExtensions.cs:127]: I don't think this is the correct approach for this. The Editor should always be set to AcceptsReturns = true

For example if the user forces the height of the editor to "500" then the editor should still accept returns 

I think what you might be running up against here are some layout behavior that's not implemented yet

Which is why things aren't resizing for you

**Review by hartez — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**